setup.py

```diff
@@ -17,12 +17,14 @@
         'future>=0.15',
         'requests>=2.6.0',
         'pytz>=2019.3',
-        'python-dateutil>=2.8.0'
+        'python-dateutil>=2.8.0',
+        'pythonsqlparser==0.1.1',
```

Comment:
I don't really like fixed requirements. Is it possible to let it float (always use the latest)?

Reply:
I vow against it as there is no guarantee that no breaking changes will be introduced between 2 versions.

This implies that pysparkling can stop working at any time, and that from one build to another something can break even though it is not related to the change introduce in pysparkling. It happened to many time before with new pylint rules for instance.

I am in favor of using `dependency~=x.y.0` for projects that use semver (when x>0), as it ensures we benefit from bugfixes but no new ,unexpected, behavior is added.

Reply:
True. `~=0.1` is better than "==0.1.1".

Reply:
For tool with a major at 0 there's the special rule of semver:

> Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.

So same here we probably want to keep a `dependency==0.y.z`

As per whether we should use `~=1.0` or `~=1.0.0`:
`~=1.0.0` will match the latest `1.1.z`, so we will have bugfixes but no new feature without explicitly modifying the version
`~=1.0` will match the latest `1.y.x`, in the case of pylint they add new rules between minor, so things would break

`~=1.0.0` appear as more resilient and to fit more our need no?

Reply:
Thanks for the explanation! I wasn't aware of the "0."-rule.

Agreed on only getting patches. However, isn't there a mechanism (maybe via Github actions?) to warn us that newer versions are available. Because in the OS world generally the latest "major.minor" and the one just before that are getting updates. Anything older is generally no longer maintained.
And I would advice against using old code.

Reply:
I agree with everything that was said here. `pythonsqlparser` is a special relationship here with pysparkling and I think a fixed or `~=` version is a good idea whereas normally I'd be against fixed versions.
In general, my rule is: for libraries like pysparkling never fix a version, but for particular apps you want to freeze all dependencies.